gpkit/constraints/set.py

```diff
@@ -1,222 +1,24 @@
-        # initializations for attributes used elsewhere
-        self.posymap = []
-
-    __str__ = _str
-    __repr__ = _repr
-    _repr_latex_ = _repr_latex_
-
-    def str_without(self, excluded=None):
-        "String representation of a ConstraintSet."
-        if not excluded:
-            excluded = ["units"]
-        lines = []
-        if "root" not in excluded:
-            excluded.append("root")
-            lines.append("")
-            root_str = self.rootconstr_str(excluded)
-            if root_str:
-                lines.append(root_str)
-        for constraint in self:
-            cstr = constraint.subconstr_str(excluded)
-            if cstr is None:
-                cstr = try_str_without(constraint, excluded)
-            if cstr[:8] != "        ":  # require indentation
-                cstr = "        " + cstr
-            lines.append(cstr)
-        return "\n".join(lines)
-
-    def latex(self, excluded=None):
-        "LaTeX representation of a ConstraintSet."
-        if not excluded:
-            excluded = ["units"]
-        lines = []
-        root = "root" not in excluded
-        if root:
-            excluded.append("root")
-            lines.append("\\begin{array}[ll] \\text{}")
-            root_latex = self.rootconstr_latex(excluded)
-            if root_latex:
-                lines.append(root_latex)
-        for constraint in self:
-            cstr = constraint.subconstr_latex(excluded)
-            if cstr is None:
-                cstr = constraint.latex(excluded)
-            if cstr[:6] != "    & ":  # require indentation
-                cstr = "    & " + cstr + " \\\\"
-            lines.append(cstr)
-        if root:
-            lines.append("\\end{array}")
-        return "\n".join(lines)
+class ConstraintSet(Constraint):
+    """
+    A Constraint with some user-facing features
 
-    def rootconstr_str(self, excluded=None):
-        "The appearance of a ConstraintSet in addition to its contents"
-        pass
+    Arguments
+    ---------
+    constraints: Iterable
+    substitutions: dict
+    """
 
-    def rootconstr_latex(self, excluded=None):
-        "The appearance of a ConstraintSet in addition to its contents"
-        pass
-
-    def subconstr_str(self, excluded=None):
-        "The collapsed appearance of a ConstraintSet"
-        pass
-
-    def subconstr_latex(self, excluded=None):
-        "The collapsed appearance of a ConstraintSet"
-        pass
-
-    def flat(self, constraintsets=True):
-        "Yields contained constraints, optionally including constraintsets."
-        for constraint in self:
-            if not isinstance(constraint, ConstraintSet):
-                yield constraint
+    def __getitem__(self, key):
+        if isinstance(key, int):
+            return list.__getitem__(self, key)
+        else:
+            variables = [Variable(**key.descr) for key in self.varkeys[key]]
+            if len(variables) == 1:
+                return variables[0]
```

Comment:
Does this need to return a list of `Variable`s instead of a list of varkeys? It seems returning `self.varkeys[key]` would solve the circular dependency issue.